python/tests/kernel/test_run_kernel.py

```diff
@@ -980,6 +981,57 @@ def kernel_with_args(real: float, imag: float) -> complex:
     assert 'unsupported return type' in str(e.value)
 
 
+def test_dataclass_does_not_support_list_type():
+
+    @dataclass
+    class TestClass:
+        x: list
+        y: int
+
+    @cudaq.kernel
+    def kernel() -> TestClass:
+        return TestClass([1, 2, 3], 4)
+
+    with pytest.raises(RuntimeError) as e:
+        cudaq.run(kernel, shots_count=1)
+    assert f"`<class 'list'>` type is not yet supported in data classes. The allowed types are: {ALLOWED_TYPES_IN_A_DATACLASS}." in str(
```

Comment:
You might even not need `ALLOWED_TYPES_IN_A_DATACLASS`. Just match the first part of the exception string.